server/aws-lsp-codewhisperer/src/language-server/localProjectContext/localProjectContextController.ts

```diff
@@ -0,0 +1,203 @@
+    }
+
+    public async init(vectorLib?: any): Promise<void> {
+        try {
+            const vecLib = vectorLib ?? (await import(path.join(LIBRARY_DIR, 'dist', 'extension.js')))
+            const root = this.findCommonWorkspaceRoot(this.workspaceFolders)
+            this._vecLib = await vecLib.start(LIBRARY_DIR, this.clientName, root)
+            LocalProjectContextController.instance = this
+        } catch (error) {
+            this.log.error('Vector library failed to initialize:' + error)
+        }
+        await this.buildIndex()
+    }
+
+    public async dispose(): Promise<void> {
+        if (this._vecLib) {
+            await this._vecLib?.clear?.()
+            this._vecLib = undefined
+        }
+    }
+
+    public async updateIndex(filePaths: string[], operation: UpdateMode): Promise<void> {
+        if (!this._vecLib) {
+            return
+        }
+
+        try {
+            await this._vecLib?.updateIndexV2(filePaths, operation)
+        } catch (error) {
+            this.log.error(`Error updating index: ${error}`)
+        }
+    }
+
+    private async buildIndex(): Promise<void> {
+        try {
+            if (this._vecLib) {
+                const sourceFiles = await this.processWorkspaceFolders(this.workspaceFolders)
+                const rootDir = this.findCommonWorkspaceRoot(this.workspaceFolders)
+                await this._vecLib?.buildIndex(sourceFiles, rootDir, 'all')
+            }
+        } catch (error) {
+            this.log.error(`Error building index: ${error}`)
+        }
+    }
+
+    public async updateWorkspaceFolders(added: WorkspaceFolder[], removed: WorkspaceFolder[]): Promise<void> {
+        try {
+            if (this._vecLib) {
+                const afterRemovals = this.workspaceFolders.filter(
+                    existing => !removed.some(removal => this.areWorkspaceFoldersEqual(existing, removal))
+                )
+
+                const merged = [...afterRemovals]
+                for (const addition of added) {
+                    if (!merged.some(existing => this.areWorkspaceFoldersEqual(existing, addition))) {
+                        merged.push(addition)
+                    }
+                }
+                await this.buildIndex()
+            }
+        } catch (error) {
+            this.log.error(`Error in updateWorkspaceFolders: ${error}`)
+        }
+    }
+
+    public async queryInlineProjectContext(
+        request: QueryInlineProjectContextRequestV2
+    ): Promise<InlineProjectContext[]> {
+        if (!this._vecLib) {
+            return []
+        }
+
+        try {
+            const resp = await this._vecLib?.queryInlineProjectContext(request.query, request.filePath, request.target)
+            return resp ?? []
+        } catch (error) {
+            this.log.error(`Error in queryInlineProjectContext: ${error}`)
+            return []
+        }
+    }
+
+    public async queryVectorIndex(request: QueryRequest): Promise<Chunk[]> {
```

Comment:
Interesting to know: how is it planned to work for [context](https://github.com/aws/aws-toolkit-vscode/blob/f1dded7edcf7d4d95410f31cfb87da208e91fb27/packages/core/src/codewhispererChat/controllers/chat/controller.ts#L564) given `getContextCommandItems` and `getContextCommandPrompt` are not exposed

Reply:
The currently exposed APIs represent what was needed for existing inline and chat use cases that we were explicitly targeting. However the others can easily be added on a case by case basis.